backend/internal/observability/service.go

```diff
@@ -0,0 +1,270 @@
+	logger    *log.Logger
+	config    *Config
+	enabled   bool
+}
+
+var (
+	serviceInstance *Service
+	serviceOnce     sync.Once
+)
+
+// GetService returns the singleton instance of the observability service.
+// Uses default configuration. For custom configuration, use InitializeWithConfig first.
+func GetService() *Service {
+	serviceOnce.Do(func() {
+		serviceInstance = NewService(nil)
+	})
+	return serviceInstance
+}
+
+// InitializeWithConfig initializes the observability service with custom configuration.
+// Must be called before GetService to take effect.
+// Returns error only if FailureMode is "strict", otherwise logs error and disables observability.
+func InitializeWithConfig(cfg *Config) (*Service, error) {
+	var svc *Service
+
+	serviceOnce.Do(func() {
+		svc = newServiceWithConfig(cfg)
+		serviceInstance = svc
+	})
+
+	if serviceInstance == nil {
+		return nil, errors.New("failed to initialize observability service")
+	}
+
+	return serviceInstance, nil
+}
+
+// NewService creates and initializes a new observability service with the given configuration.
+func NewService(cfg *Config) *Service {
+	svc := newServiceWithConfig(cfg)
+	return svc
+}
+
+// newServiceWithConfig creates and initializes a new observability service.
+func newServiceWithConfig(cfg *Config) *Service {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, loggerComponentName))
+
+	// Use default config if none provided
+	if cfg == nil {
+		cfg = DefaultConfig()
+	}
+
+	// Check if observability is disabled
+	if !cfg.Enabled {
+		logger.Info("Observability service is disabled by configuration")
+		return &Service{
+			publisher: nil,
+			logger:    logger,
+			config:    cfg,
+			enabled:   false,
+		}
+	}
+
+	logger.Info("Initializing observability service",
+		log.String("outputType", cfg.Output.Type))
+
+	// Create event bus (no queue needed)
+	pub := publisher.NewCategoryPublisher()
+
+	svc := &Service{
+		publisher: pub,
+		logger:    logger,
+		config:    cfg,
+		enabled:   true,
+	}
+
+	// Initialize default subscribers
+	svc.initializeDefaultSubscribers()
+
+	logger.Info("Observability service initialized successfully")
+	return svc
+}
+
+// initializeDefaultSubscribers sets up the default subscribers for analytics events.
+func (s *Service) initializeDefaultSubscribers() {
+	// Create formatter based on config
+	var fmtr formatter.Formatter
+	switch s.config.Output.Format {
+	case "json", "":
+		fmtr = jsonformatter.NewJSONFormatter()
+	default:
+		fmtr = jsonformatter.NewJSONFormatter()
+		s.logger.Warn("Unknown output format, using JSON",
+			log.String("format", s.config.Output.Format))
+	}
+
+	// Create adapter based on config
+	var adptr adapter.OutputAdapter
+	var err error
+
+	switch s.config.Output.Type {
+	case "file", "":
+		// Use file path from config or default
+		analyticsFile := s.config.Output.File.Path
+		if analyticsFile == "" || analyticsFile == "${THUNDER_HOME}/logs/analytics/analytics.log" {
```

Comment:
Are we expecting the users to set the file path in absolute manner with ${THUNDER_HOME}?